File: 0931-minimum-falling-path-sum/0931-minimum-falling-path-sum.ts
function minFallingPathSum(matrix: number[][]): number {

    const n = matrix.length;

    for (let i = 1; i < n; i++) {

        for (let j = 0; j < n; j++) {

            let up = matrix[i - 1][j];

            let left = j > 0
                ? matrix[i - 1][j - 1]
                : Number.MAX_SAFE_INTEGER;

            let right = j < n - 1
                ? matrix[i - 1][j + 1]
                : Number.MAX_SAFE_INTEGER;

            matrix[i][j] += Math.min(up, left, right);
        }
    }

    let ans = Number.MAX_SAFE_INTEGER;

    for (let j = 0; j < n; j++) {
        ans = Math.min(ans, matrix[n - 1][j]);
    }

    return ans;
}

File: 0931-minimum-falling-path-sum/0931-minimum-falling-path-sum.java
class Solution {
    public int minFallingPathSum(int[][] matrix) {

        int n = matrix.length;

        for (int i = 1; i < n; i++) {

            for (int j = 0; j < n; j++) {

                int up = matrix[i - 1][j];

                int left = (j > 0)
                        ? matrix[i - 1][j - 1]
                        : Integer.MAX_VALUE;

                int right = (j < n - 1)
                        ? matrix[i - 1][j + 1]
                        : Integer.MAX_VALUE;

                matrix[i][j] += Math.min(up, Math.min(left, right));
            }
        }

        int ans = Integer.MAX_VALUE;

        for (int val : matrix[n - 1]) {
            ans = Math.min(ans, val);
        }

        return ans;
    }
}